src/gameboy/src/gameboy.cpp

```diff
@@ -2,9 +2,15 @@
 
 namespace gameboy {
 
-Gameboy::Gameboy(std::string rom_path) {
+Gameboy::Gameboy(std::string rom_path, bool is_recording,
+                 std::string json_file_name) {
 	cartridge = std::make_unique<Cartridge>(rom_path);
-	controller = std::make_unique<Controller>();
+
+	if (!is_recording)
+		controller = std::make_unique<Controller>();
+	else
+		controller = std::make_unique<RecordingController>(json_file_name);
```

Comment:
Minor nit: Is there a specific reason this condition is inverted? I think it reads better without the negation.

Reply:
Nope no reason. I'll correct it right away